planner/core/logical_plan_builder.go

```diff
@@ -254,6 +258,14 @@ func (b *PlanBuilder) buildAggregation(ctx context.Context, p LogicalPlan, aggFu
 		schema4Agg.Append(newCol)
 		names = append(names, p.OutputNames()[i])
 	}
+	hasGroupBy := len(gbyItems) > 0
+	for i, aggFunc := range plan4Agg.AggFuncs {
+		err := aggFunc.UpdateNotNullFlag4RetType(hasGroupBy, allAggsFirstRow)
```

Comment:
Need we put this logic directly to the `NewAggFuncDesc`? Many other places also build aggregate plan. Should those places also consider to update the `NotNull` flag?

Reply:
We need to know information about other aggregate functions before we can update the NotNull flag of an aggregate function, so we cannot do it in `NewAggFuncDesc`?